fiftyone/core/stages.py

```diff
@@ -1064,16 +1090,24 @@ def _get_filter_list_field_pipeline(
             }
         )
 
+    if hide_result:
+        pipeline.append({"$unset": new_field})
+
     return pipeline
 
 
 def _get_filter_frames_list_field_pipeline(
-    filter_field, filter_arg, only_matches=False, hide_frames=False
+    filter_field,
+    new_field,
+    filter_arg,
+    only_matches=False,
+    hide_frames=False,
```

Comment:
I must admit that I'm a bit intimidated by the `hide_frames`, `hide_results`, `squash_frames`, `include_frames`, `_needs_frames` business in the codebase at the moment. I don't quite understand it all.

So for now I'll just verify that the API works :)